backend/de.metas.payment.sepa/base/src/main/java/de/metas/payment/sepa/sepamarshaller/impl/SEPAVendorCreditTransferMarshaler_Pain_001_001_03_CH_02.java

```diff
@@ -1003,12 +1015,59 @@ private String createInfo(@NonNull final I_SEPA_Export_Line line)
 		}
 		if (line.getC_BP_BankAccount_ID() > 0)
 		{
-			sb.append(" @C_BP_BankAccount_ID@ ");
-			sb.append(line.getC_BP_BankAccount().getDescription());
+			if (line.getC_BP_BankAccount().getDescription() != null)
+			{
+				sb.append(" @C_BP_BankAccount_ID@ ");
+				sb.append(line.getC_BP_BankAccount().getDescription());
+			}
+			if (line.getC_BP_BankAccount().getName() != null)
+			{
+				sb.append(" @Name@ ");
+				sb.append(line.getC_BP_BankAccount().getName());
+			}
+			if (line.getC_BP_BankAccount().getIBAN() != null)
+			{
+				sb.append(" @IBAN@ ");
+				sb.append(line.getC_BP_BankAccount().getIBAN());
+			}
+		}
+		if (line.getStructuredRemittanceInfo() != null)
+		{
+			sb.append(" @StructuredRemittanceInfo@ ");
+			sb.append(line.getStructuredRemittanceInfo());
+		}
+		if (line.getRecord_ID() > 0)
+		{
+			sb.append(" @C_Invoice_ID@ ");
+			sb.append(getInvoiceIDByRecordID(line));
+			sb.append(" @Line@ ");
+			sb.append(getLineNoByRecordID(line));
 		}
+
```

Comment:
Where is the QR Reference?

Reply:
the QR Refence is here:
```java
if (line.getStructuredRemittanceInfo() != null)
		{
			sb.append(" @StructuredRemittanceInfo@ ");
			sb.append(line.getStructuredRemittanceInfo());
		}
```
I asked cris if I shouldn't change the name of the string there since it would be confusing. But that is the name of the column there so changing it would be even more confusing lateron. So I think we could add a comment to show that this is the QR reference?

Reply:
I changed it to take from the payselectionline like the other methods that I introduced earlier.